shell/server/core.js

```diff
@@ -176,17 +176,40 @@ class NotificationHandle {
 }
 
 class PersistentUiViewImpl {
-  constructor(persistentMethods) {
+  constructor(persistentMethods, grain) {
+    this._grain = grain;
     _.extend(this, persistentMethods);
   }
 
+  getViewInfo() {
+    const viewInfo = this._grain.cachedViewInfo;
+    return inMeteor(() => {
+      const pkg = Packages.findOne({ _id: this._grain.packageId });
+      const manifest = pkg.manifest || {};
+      if (!viewInfo.appTitle) {
+        viewInfo.appTitle = manifest.appTitle || {};
+      }
+
+      if (!viewInfo.grainIconUrl) {
+        // TODO(security, now): Do we need to do some kind of filtering here to prevent
```

Comment:
This TODO wouldn't be caught by our release script, FWIW. You need to do `TODO(now) TODO(security)`, or just say that `TODO(now)` takes precedence.